test/dlc_tests/container_tests/bin/pytorch_tests/testTorchdataDev

```diff
@@ -4,13 +4,36 @@ HOME_DIR=/test
 BIN_DIR=${HOME_DIR}/bin
 LOG_DIR=${HOME_DIR}/logs
 
-TORCHDATA_DEV_BRANCH_PREFIX="torchdata_pt_"
-TORCHDATA_DEV_REPO="https://github.com/ShiboXing/data.git"
-TORCHDATA_TEST_SCRIPT="data/test/test_remote_io.py"
-VERSION=$(python -c "import torch; print(torch.__version__.split('+')[0])")
+TORCHDATA_REPO="https://github.com/pytorch/data.git"
+TORCHDATA_TEST_SCRIPT="test/test_remote_io.py"
+TORCH_VERSION=$(python -c "import torch; print(torch.__version__)")
+TORCHDATA_RELEASE_TAG=$(python -c "import torchdata; torchdata_versions = torchdata.__version__.split('.'); print(f'release/{torchdata_versions[0]}.{torchdata_versions[1]}')")
 
-echo "Fetch Torchdata S3 IO test script."
-git clone -b ${TORCHDATA_DEV_BRANCH_PREFIX}${VERSION} ${TORCHDATA_DEV_REPO}
+## using the commit requires user.email and user.name.. 
+echo adding temp git config user email and name
+git config --global user.email "you@example.com"
+git config --global user.name "Your Name"
+
+echo cloning torchdata repo
+git clone ${TORCHDATA_REPO}
+cd data
+
+## PyTorch 1.11 DLC was built with TorchData 0.4.0 which is meant for PyTorch 1.12 and not PT 1.11
+## This fails the Torchdata 0.4.0 test_remote_io.py at test_http_reader_iterdatapipe due to 
+## Torch not knowing "auth" in 1.11. We need to use TorchData 0.3.0 Branch for testing PyTorch 1.11 DLC
+if [[ ${TORCH_VERSION} == *"1.11"* ]]
```

Comment:
string comparison of versions is prone to error. If a special case needs to be added for PT 1.11, I suggest doing this via python in the test function and passing in the overridden release version as an optional script argument. This will also keep the underlying test script cleaner and allow the the caller to override the release tag if needed.

Reply:
In this case, considering the file is testTorchdata"Dev" and is not staying once PT 1.11 goes out of support. I would prefer to leave this as is. If need be, I can make it more exact, rather than using "contains". we are not gaining anything by me converting all this to Python for a framework version, soon to go out of service.

Reply:
Kept the bash, but updated to 1.11.0 with no wildcards making more exact. this is under the assumption that PT will not rev 1.11.0 to 1.11.1.

Reply:
I took a hybrid approach to this. the string is compared in Python and it provides a bool for the if statement.